lib/matplotlib/widgets.py

```diff
@@ -904,14 +914,34 @@ def set_val(self, val):
             xy[2] = .75, val[1]
             xy[3] = .75, val[0]
             xy[4] = .25, val[0]
+
+            hbottom_pos = self._handles[0].get_ydata()
+            htop_pos = self._handles[1].get_ydata()
+            if hbottom_pos != val[0]:
+                hbottom_pos[0] = val[0]
```

Comment:
are all these checks necessary or can we we just go ahead and set them?

Reply:
Also *I think* that the more technically correct way to get the position as a scalar (which is how you're using it here) 
would be with:
```python
             hbottom_pos = self._handles[0].get_ydata() [0]
```

because `get_{x,y}data` returns an array (length 1 in this case) and then you're comparing an array to scalar adn asking if theyre equal

Reply:
The RangleSlider._update_val_from_pos in an array and not just a scalar (for some reason). 

I originally wrote the code to get a scalar and got this error:
![image](https://user-images.githubusercontent.com/28690153/160283927-3fb1cdf9-db17-4d5e-8f15-95f93c6368d1.png)

With that said, I am no longer doing the comparison, so I was able to remove that line completely and just use the value passed into the set_val function.

I re-tested my two examples and everything is looking great!